### DISCUSSION TITLE
Adding Breaks to Pickup and Delivery Solution

### DISCUSSION THREAD

**Discussion by remeshx** [google/or-tools#3929] [Routing (and legacy CP) questions]
Version: v9.5 
Language: Python

I am new to OR-Tools and have successfully created a solution for pickup and delivery problems with time windows and breaks using the following code:

```python       
      node_visit_transit = [0] * routing.Size()
      for index in range(routing.Size()):
          node = manager.IndexToNode(index)
          node_visit_transit[index] = data['service_time'][node]
      
      break_intervals = {}
      for v in range(manager.GetNumberOfVehicles()):
          if not data['vehicle_breaks'][v] is None:
              breakStartTime = data['vehicle_breaks'][v][0]
              breakEndTime = data['vehicle_breaks'][v][1]
              breakDuration = breakEndTime- breakStartTime
              break_intervals[v] = [
                  routing.solver().FixedDurationIntervalVar(
                      breakStartTime,  # start min
                      breakStartTime + 500,  # start max
                      breakDuration,  # duration: 10 min
                      False,  # optional: no
                      f'Break for vehicle {v}')
              ]
     
        time_dimension.SetBreakIntervalsOfVehicle(
            break_intervals[v],  # breaks
            v,  # vehicle index
            node_visit_transit)
```
However, although the breaks are correctly placed between the trips, they do not consider the pickup and delivery constraints. In other words, when a vehicle picks up a passenger, the break should not be allowed while the passenger is onboard. Unfortunately, the solution currently only schedules breaks when passengers have been picked up but not yet dropped off.
is that expected behavior? 
How can one enforce the solver to take into account the pickup and drop-off constraints for breaks and only schedule a break when the vehicle's capacity is at its maximum value (i.e., when it does not have any passengers onboard)?"

**Comment by Mizux** [comment]
Break don't know you have a capacity/load dimension.

If you only have a capacity of 1, you can set the SlackVar to any pickup point to 0 so solver won't be able to add a break after a pickup since the SlackVar (i.e. waiting time) is zero.
